enhancements/agent-installer/automated-workflow.md

```diff
@@ -0,0 +1,613 @@
+[NMStateConfig](https://github.com/openshift/assisted-service/blob/master/api/v1beta1/nmstate_config_types.go)
+resources to specify static network settings for the hosts. Although
+there are similar fields in `install-config.yaml`, they are only
+available when the platform setting is `baremetal`, and the
+agent-based installer must support other platforms.
+
+We anticipate using
+[Butane](https://github.com/coreos/butane/blob/main/docs/config-openshift-v4_10.md)
+file in the future for specifying storage settings, although that
+decision is subject to change in a future enhancement.
+
+The `agent create-image` command will produce a single image as output, to
+save storage space and reduce management complexity. The assisted
+installer service can already generate images with boot scripts that
+recognize which host is booting and apply host-specific network
+settings and roles (such as running different services) accordingly,
+and we will reuse that technique (and code).
+
+The image generated by the `agent create-image` command will be an RHCOS
+live ISO with embedded Ignition config containing systemd units to
+start the assisted service, agent, network, etc. running in containers
+pulled from the OpenShift release payload. Using RHCOS ensures that
+the same OS that will run OpenShift boots on the hosts. Using
+containers in the release payload means that when the payload is
+mirrored all of the tools needed to run the agent based deployment
+workflows will be present in the mirror.  More details about the
+various services follow below.
+
+The image generated by the `agent create-image` command will also
+contain the data in the input `install-config.yaml`,
+`agent-config.yaml`, and other files. That data may be converted to
+another format (we anticipate using the ZTP CRDs, since those map to
+the assisted-service API calls). The intermediate format will be
+considered another layer of control for the cluster creator, just as
+the manifests written by the `create manifests` sub-command are. The
+cluster creator will be able to generate those intermediate files with
+the sub-command `agent create-manifests`.
+
+One host, designated as "node0", will be configured to run the
+assisted service and a client that uses the assisted service REST API
+to automate the deployment. Because we must support single-node and
+compact deployments, the node selected as node0 must always be
+configured as a control plane node. Selecting other control plane
+nodes may be left to the user.
+
+All hosts will run the assisted agent, configured to talk to the
+service on node0.
+
+When all of the hosts are booted with the image, the client on node0
+will use the assisted service to trigger the deployment of the cluster
+(including applicable pre-flight validations) without further
+interaction from the user.
+
+The `agent create-image` command will produce as output credentials to
+access the cluster that is about to be created. This allows a client,
```

Comment:
Do we know how to get the credentials into the installed cluster in a supported way?

Reply:
We're still working on that.